src/vos/vos_dtx.c

```diff
@@ -1161,6 +1164,13 @@ vos_dtx_check_availability(daos_handle_t coh, uint32_t entry,
 	}
 
 	if (intent == DAOS_INTENT_PURGE) {
+		d_iov_set(&kiov, &DAE_XID(dae), sizeof(DAE_XID(dae)));
+		d_iov_set(&riov, NULL, 0);
+
+		rc = dbtree_lookup(cont->vc_dtx_active_hdl, &kiov, &riov);
+		D_ASSERTF(rc == 0, "DTX "DF_DTI" (%u) should be in active table: %d\n",
```

Comment:
is it a good idea for this to be an assertion? what happens if the entry is not in active table? can we get into that situation?

Reply:
From logic perspective, this situation should never happen. But if it really happened, it must be some DAOS logic bug. Now it is suspected that because of such bad case, related vos aggregation was blocked. That is why I add the assertion here.